### PR TITLE
Refactor sb

### PR DESCRIPTION
**Pull request by stellakunzang**
Refactored long, slow methods. 
Implemented modules to reduce repetition. 
Fixed some minor formatting issues. 
Added tests that were missing.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,20 @@
...
+
+* self.from_csv (which instantiates objects in the respective class by reading the CSV)
+
+* self.all (which gives us access within the class to all of the instances by means of an instance variable)
+
+* self.find_by (which allows us to search using game_id)
+
+Using inheritance for these methods was pretty straightforward since they were exactly the same across the three classes and were being used to set _state_ and _behavior_. The repetition we saw in methods across the classes was not as straightforward and we decided to write modules since they were strictly dealing with _behavior_.
+
+...
```
Comment (stellakunzang) [inline README.md]: Work in progress. If there is time before approving the PR, I'll finish and push up changes for review.

File: lib/game.rb [hunk 2]
```diff
@@ -52,37 +50,35 @@ def self.average_goals_by_season
   def self.average_goals_by(hoa_team)
     divide_hash_values(:goals, :games_played, games_goals_by(hoa_team))
   end
-#deliverable
```
Comment (stellakunzang) [inline lib/game.rb:55]: it seemed ok at this point to remove these. If that's not ok, I'll put them back.

File: lib/game_team.rb [hunk 3]
```diff
@@ -12,71 +12,38 @@ def self.find_by_team(team_id)
...
-  end
-#deliverbale
-  def self.wins_by_coach(season_id)
-    wins_by_coach_by_season = Hash.new { |hash, key| hash[key] = 0 }
-    results_by_coach(season_id).each do |coach, results|
-      results.each do |result|
-        wins_by_coach_by_season[coach] += 1 if result == "WIN"
-      end
-    end
-    wins_by_coach_by_season
```
Comment (stellakunzang) [inline lib/game_team.rb:68]: Booyah!

File: test/game_team_test.rb [hunk 4]
```diff
@@ -96,23 +96,12 @@ def test_it_can_find_percentage_ties
...
-  end
-
-  def test_it_can_find_total_games_coached
-    assert_equal ({"Willie Desjardins"=>2, "Darryl Sutter"=>2, "Claude Julien"=>2, "Michel Therrien"=>1, "Joel Quenneville"=>1, "Jared Bednar"=>1, "Mike Babcock"=>2, "Bruce Cassidy"=>1, "Lindy Ruff"=>2}), GameTeam.total_games_coached(2016030134)
-  end
-
-  def test_it_can_count_wins_by_coach
-    assert_equal ({"Claude Julien"=>2, "Joel Quenneville"=>1, "Mike Babcock"=>1, "Bruce Cassidy"=>1, "Lindy Ruff"=>2}), GameTeam.wins_by_coach(2016030134)
-  end
-
```
Comment (stellakunzang) [inline test/game_team_test.rb:114]: boom

File: lib/game.rb (hunk 2 above)
Comment (foleymichelle9) [reply lib/game.rb:55]: Agreed I think it looks better without them

File: lib/game_team.rb (hunk 3 above)
Comment (foleymichelle9) [reply lib/game_team.rb:68]: noiice 😎

File: test/stat_tracker_test.rb [hunk 5]
```diff
@@ -92,7 +90,7 @@ def test_it_can_return_team_name_with_most_tackles
   def test_it_can_return_team_name_with_fewest_tackles
     assert_equal "Washington Spirit FC", @stat_tracker.fewest_tackles("20122013")
   end
-  #michelle end
```
Comment (foleymichelle9) [inline test/stat_tracker_test.rb:95]: Thanks for catching that!

**Review by foleymichelle9 — APPROVED** [review]
Everything looks great Stella! Your refactors are golden 🥇

File: README.md (hunk 1 above)
Comment (foleymichelle9) [reply README.md]: Stella! This is so in depth and concise. Thank you for taking the time to write this and for the amount of detail you included!

File: README.md [hunk 6]
```diff
@@ -1,3 +1,24 @@
...
+Our modules are called Gameable, Hashable, and Winable. Hashable was extremely useful and showed us all the value of using hashes, and moreover a hash within a hash, to reduce the need for nested iteration. It allowed us to, in a very direct way, iterate over our data to calculate the desired values, and then to divide those values to obtain a percentage. What had originally taken many lines and helper methods was reduced exponentially by implementing the Hashable module and extensively refactoring to utilize the methods contained therein.
+
+The Gameable module allowed us to cutdown on repetition specifically within the Game class methods. We were running very similar code within many methods and found that by pulling the methods out into a module and passing in an argument we were able to cut down on repetition within our code drastically.The Winable module was also designed to reduce repetition in the Game class.
+
+Overall, our design philosophy was governed by a collective desire to reduce unnecessary repetition and excessive nested iteration, and produce elegant, clean, concise Ruby code. We found that inheritance and modules were powerful new tools to accomplish this goal.
```
Comment (perryr16) [inline README.md:24]: well played stella

File: lib/game_team.rb (hunk 3 above)
Comment (perryr16) [reply lib/game_team.rb:68]: stellar

File: lib/game_team.rb [hunk 7]
```diff
@@ -162,46 +128,28 @@ def self.fewest_goals_scored(team_id)
...
+
   def self.rival_id(team_id)
-    team_id = team_id
-    record_length = {}
-    opponents_records(team_id).map do |team_id, record|
-      record_length[team_id] = record.length
-    end
-    opponents = opponents_records(team_id).keys
-    opponents.max_by {|opponent| (opponents_wins(team_id)[opponent].to_f / record_length[opponent].to_f).round(2)}
+    opponents = divide_hash_values(:wins, :games_played, opponents_records(team_id))
```
Comment (perryr16) [inline lib/game_team.rb:150]: tons of changes. loving all the red

**Review by perryr16 — COMMENTED** [review]
approved -rp

**Pull request merged**